### PR TITLE
Major refactoring to handle dependencies and other things

### PR DESCRIPTION
**Pull request by jrmi**
- Rename some nodes (with deprecation warning)
- Add Case node (New functionality)
- Add replay message method on channel (to prepare introspection and web client)
- Event on channel status change (to prepare introspection and web client)
- Better documentation (A little)
- Better tests
- Better graph generation

### CODE REVIEW

File: pypeman/channels.py [hunk 1]
```diff
@@ -41,36 +38,35 @@ class Break(Exception):
...
     pass
 
 
 class BaseChannel:
     STARTING, WAITING, PROCESSING, STOPPING, STOPPED  = range(5)
 
-    dependencies = [] # List of module requirements
-
-    def __init__(self, name=None, parent_channel=None, loop=None, force_msg_order=True, message_store_factory=None):
+    def __init__(self, name=None, parent_channel=None, loop=None, message_store_factory=None):
```
Comment (mhcomm) [inline pypeman/channels.py]: KF: force_msg_order? is this now always true or false or handled somewher else?

File: pypeman/channels.py [hunk 2]
```diff
@@ -237,36 +259,75 @@ def process(self, msg):
         else:
             return msg
 
+    @asyncio.coroutine
+    def replay(self, msg_id):
+        """
+        This method allow you to replay a message from channel `message_store`.
```
Comment (mhcomm) [inline pypeman/channels.py]: allow -> allows

File: pypeman/channels.py [hunk 3]
```diff
@@ -237,36 +259,75 @@ def process(self, msg):
...
+            elif isinstance(node, Case):
+                for i, c in enumerate(node.cases):
+                    print(prefix + '|c%s\\' % i)
+                    c[1].graph(prefix='|  ' + prefix)
+                    print(prefix + '|<--')
             else:
                 print(prefix + '|-' + node.name)
 
-    def graph_dot(self, previous='', end=''):
+    def graph_dot(self, end=''):
```
Comment (mhcomm) [inline pypeman/channels.py]: always prints to stdout. shouldn't we do either:
print with a file parameter or just return list of strings, so the function can be reused

File: pypeman/channels.py [hunk 4]
```diff
@@ -356,47 +416,6 @@ def handle(self, msg):
         return result
 
 
-class HttpChannel(BaseChannel):
-    """ Channel that handle http messages.
```
Comment (mhcomm) [inline pypeman/channels.py]: handle -> handles

File: pypeman/commands.py [hunk 5]
```diff
@@ -116,49 +107,34 @@ def start(reload: 'Make server autoreload (Dev only)'=False,
 
 
 @begin.subcommand
-def graph(dot: "Make dot compatible output"=False):
-    """ Show channel graph """
+def graph(dot: "Make dot compatible output (Can be see with http://ushiroad.com/jsviz/)"=False):
```
Comment (mhcomm) [inline pypeman/commands.py]: see -> seen or perhaps better: 'viewed' / 'visualized'

File: pypeman/contrib/time.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import asyncio
+import datetime
+
+from pypeman import channels, message
+
+from aiocron import crontab
+
+class TimeChannel(channels.BaseChannel):
```
Comment (mhcomm) [inline pypeman/contrib/time.py]: better name might be TimedChannel 
TimeChannel sounds aas if the contesnts of the channel were a time

File: pypeman/helpers/lazyload.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        for dep in deps:
+            if dep in msg:
+                found = True
+                break
+
+        if not found:
+            print("IMPORT ERROR...")
+            raise
+
+        print("%s module not activated" % module)
```
Comment (mhcomm) [inline pypeman/helpers/lazyload.py]: all error prints should be to stderr not not be mixed with stdout.
that's kind of common practice for most tools

File: pypeman/tests/test_channel.py [hunk 8]
```diff
@@ -311,14 +370,47 @@ def test_memory_message_store(self):
...
+
+        self.assertEqual(len(msg_stored), 2, "Should be 3 messages in store!")
+
+        self.loop.run_until_complete(chan.replay(msg_stored[0]['id']))
+
+        msg_stored = list(chan.message_store.search())
+        self.assertEqual(len(msg_stored), 3, "Should be 3 messages in store!")
+
     def test_file_message_store(self):
         """ We can store a message in FileMessageStore """
```
Comment (mhcomm) [inline pypeman/tests/test_channel.py]: same comment in previous test. 
It should be different, as the first line of a test docstring id displayed during testing.
Easier to find the correct function if unique

File: requirements_test.txt [hunk 9]
```diff
@@ -0,0 +1,5 @@
+# Mandatory
```
Comment (mhcomm) [inline requirements_test.txt]: just to understand?
what happens if certain dependencies are not isntalled. 
Are the tests marked as skipped or as failed?

File: pypeman/channels.py (hunk 1 above)
Comment (jrmi) [inline pypeman/channels.py]: Always true as I don't know how to disabled it for now

File: pypeman/channels.py (hunk 2 above)
Comment (jrmi) [inline pypeman/channels.py]: Ok

File: pypeman/channels.py (hunk 3 above)
Comment (jrmi) [inline pypeman/channels.py]: I was thinking about using shell pipe instead as i don't really see other usage of this command.

File: pypeman/channels.py (hunk 4 above)
Comment (jrmi) [inline pypeman/channels.py]: Ok

File: pypeman/commands.py (hunk 5 above)
Comment (jrmi) [inline pypeman/commands.py]: Ok

File: pypeman/contrib/time.py (hunk 6 above)
Comment (jrmi) [inline pypeman/contrib/time.py]: Ok or CronChannel or better PeriodicChannel  ?

File: pypeman/helpers/lazyload.py (hunk 7 above)
Comment (jrmi) [inline pypeman/helpers/lazyload.py]: I agree, but I'll make a specific commit for all change in one row

Comment (jrmi) [inline pypeman/helpers/lazyload.py]: In next PR...

File: requirements_test.txt (hunk 9 above)
Comment (jrmi) [inline requirements_test.txt]: As skipped. This is only for tox validation. If you execute test in your custom environment and some dependency are missing, a message show the skipped test. See line 205 of file tests/test_node.py

**Pull request merged**